internal/models/transformer.go

```diff
@@ -25,15 +28,33 @@ func Transform(ctx context.Context, model interface{}) error {
 	return nil
 }
 
-func generateFingerprint(ctx context.Context, sl mold.StructLevel) error {
-	logger := ctxval.Logger(ctx)
+func generateFingerprints(ctx context.Context, sl mold.StructLevel) error {
 	pk := sl.Struct().Interface().(Pubkey)
 	pkey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(pk.Body))
 	if err != nil {
-		return fmt.Errorf("unable to calculate fingerprint for %s: %w", pk.Name, err)
+		return fmt.Errorf("unable to parse public key %s: %w", pk.Name, err)
 	}
-	pk.Fingerprint = ssh.FingerprintSHA256(pkey)
+
+	pk.Type = pkey.Type()
+	pk.Fingerprint = strings.TrimLeft(ssh.FingerprintSHA256(pkey), "SHA256:") + "="
+	pk.FingerprintLegacy = strings.TrimLeft(ssh.FingerprintLegacyMD5(pkey), "MD5:")
+
+	parsedCryptoKey := pkey.(ssh.CryptoPublicKey)
+	pub := parsedCryptoKey.CryptoPublicKey() //TODO
+	der, err := x509.MarshalPKIXPublicKey(pub)
+	if err != nil {
+		return fmt.Errorf("failed to calculate legacy fingerprint: %w", err)
+	}
+	md5sum := md5.Sum(der) //#nosec
+	// TODO optimize this
+	hexarray := make([]string, len(md5sum))
+	for i, c := range md5sum {
+		hexarray[i] = hex.EncodeToString([]byte{c})
+	}
+	result := strings.Join(hexarray, ":")
+
+	pk.FingerprintPEM = result
+
```

Comment:
The code @ezr-ondrej wrote was moved here essentially, where it belongs if we want to go the pre-calculated path. This is where calculation of the original fingerprint was.